### PR TITLE
Add extras links to some more EMR Operators and Sensors

### LINKED ISSUE #29939: AWS EMR Operators: Add Log URI in task logs to speed up debugging
### Description

Airflow is widely used to launch, interact and submit jobs on AWS EMR Clusters. Existing EMR operators do not provide links to the EMR logs (Job Flow/Step logs), as a result in case of failures the users need to switch to EMR Console or go to AWS S3 console to locate the logs for EMR Jobs and Steps using the job_flow_id available in the EMR Operators and in Xcom.

It will be really convenient and help with debugging if the EMR log links are present in Operator Task logs, it will obviate the need to switch to AWS S3 or AWS EMR  consoles from Airflow and lookup the logs using job_flow_ids. It will be a nice improvement for the developer experience.

LogUri for Cluster is available in [DescribeCluster](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/emr/client/describe_cluster.html)

LogFile path for Steps in case of failure is available in [ListSteps](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/emr/client/list_steps.html) 

### Use case/motivation

Ability to go to EMR logs directly from Airflow EMR Task logs.

### Related issues

N/A

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by ferruzzi**
**Overview**
Adds some "extras links" linking back to the job logs for a couple of EMR Operators and Sensors and removes a couple of out of date comments.

**Testing**
Passes `breeze static-checks --all-files` and `breeze build-docs`

closes: https://github.com/apache/airflow/issues/29939

cc: @o-nikolas @vincbeck @vandonr-amz @syedahsn

### CODE REVIEW

File: airflow/providers/amazon/aws/links/emr.py [hunk 1]
```diff
@@ -33,3 +40,23 @@ class EmrLogsLink(BaseAwsLink):
...
+
+
+def get_log_uri(
+    *, cluster: dict[str, Any] | None = None, emr_client: boto3.client = None, job_flow_id: str | None = None
+) -> str:
+    """
+    Retrieves the S3 URI to the EMR Job logs.  Requires either the output of a
+    describe_cluster call or both an EMR Client and a job_flow_id to look it up.
+    """
+    if not exactly_one(cluster, emr_client and job_flow_id):
```
Comment (ferruzzi) [inline airflow/providers/amazon/aws/links/emr.py]: Could also do an XOR `bool(cluster) ^ bool(emr_client and job_flow_id)` here, but the `exactly_one` helper was handy and more clear IMHO.

Would parens make it more clear as `if not exactly_one(cluster, (emr_client and job_flow_id)):` or is this good?

File: airflow/providers/amazon/aws/operators/emr.py [hunk 2]
```diff
@@ -597,7 +608,7 @@ class EmrCreateJobFlowOperator(BaseOperator):
     template_ext: Sequence[str] = (".json",)
     template_fields_renderers = {"job_flow_overrides": "json"}
     ui_color = "#f9c915"
-    operator_extra_links = (EmrClusterLink(),)
+    operator_extra_links = (EmrLogsLink(),)
```
Comment (pankajastro) [inline airflow/providers/amazon/aws/operators/emr.py]: why we are removing `EmrClusterLink`?

File: airflow/providers/amazon/aws/links/emr.py (hunk 1 above)
Comment (vincbeck) [reply airflow/providers/amazon/aws/links/emr.py]: lgtm

File: airflow/providers/amazon/aws/operators/emr.py (hunk 2 above)
Comment (ferruzzi) [reply airflow/providers/amazon/aws/operators/emr.py]: There is a very good reason for it.   I messed up.  Adding it back now, thanks for the catch.

**Comment by ferruzzi** [comment]
I will never understand how thee tests will pass in breeze but not in the CI.   I have a busy day today; I'll have a look at them this afternoon or tomorrow.

**Review by vincbeck — APPROVED** [review]

File: airflow/providers/amazon/aws/links/emr.py (hunk 1 above)
Comment (ferruzzi) [reply airflow/providers/amazon/aws/links/emr.py]: Yeah, thanks.   The more I look at it the more I think the "clarifying" parens are even more confusing.  I'll leave it like this.

**Review by o-nikolas — APPROVED** [review]

**Pull request merged**